include/npcomp/Dialect/TCP/IR/TCPOps.td

```diff
@@ -85,27 +114,64 @@ def TCP_GetGlobalMemrefOp : TCP_Op<"get_global_memref"> {
   let verifier = "return ::verify$cppClass(*this);";
 }
 
-// TODO: Change to a more principled error handling mechanism.
-// This op probably doesn't need to exist eventually.
-// This op is also not correctly modeled right now, since it itself doesn't
-// produce the error in practice. The ops like shape.broadcast itself, when
-// lowered, immediately produce errors.
-// TODO: This should eventually be moved to a shape dialect.
-def TCP_ShapeObserveErrorOp : TCP_Op<"shape_observe_error", []> {
-  let summary = "Observes the fact that a shape might be an error.";
+//===----------------------------------------------------------------------===//
+// Ops related to shapes.
+//===----------------------------------------------------------------------===//
+// TODO: These belong in a shape-related dialect.
+
+def TCP_ShapedResultsOp : TCP_Op<"shaped_results", [
```

Comment:
FYI - Really glad you landed on this formulation. This is how I wish I had started with IREE's shapex.tie_shape op, and likely we will want to adapt that to something like this.